File: rei-do-chopp-site/src/app/restockings/models/requests/restocking-list.request.ts
import { PaginationRequest } from "../../../shared/models/requests/pagination.request";

export class RestockingListRequest extends PaginationRequest {
  InitialDate: string;
  FinalDate: string;
  ProductsIds: Array<number>

  constructor(params: Partial<{ InitialDate: Date, FinalDate: Date, ProductsIds: Array<number> }> & Partial<PaginationRequest>) {
    super(params);
    this.InitialDate = params.InitialDate.toISOString() || null;
    this.FinalDate = params.FinalDate.toISOString() || null;
    this.ProductsIds = params.ProductsIds || [];
  }
}

File: rei-do-chopp-api/ReiDoChopp.DataTransfer/Restockings/Requests/RestockingListRequest.cs
using System;
using ReiDoChopp.Domain.Utils.Enums;
using ReiDoChopp.DataTransfer.Pagination.Responses;
using ReiDoChopp.DataTransfer.Pagination.Requests;

namespace ReiDoChopp.DataTransfer.Restockings.Request
{
    public class RestockingListRequest : PaginationRequest
    {
        public int? Id { get; set; }
        public DateTime? InitialDate { get; set; }
        public DateTime? FinalDate { get; set; }
        public int? UserId { get; set; }
        public double? ProductSumValue { get; set; }
        public double? TotalFees { get; set; }
        public int[] ProductsIds { get; set; }

        public RestockingListRequest() : base(pageSize: 10, ordenationField: "Id", ordenationType: OrdenationTypeEnum.DESC) {}
    }
}